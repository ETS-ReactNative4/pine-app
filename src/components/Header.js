import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import PropTypes from 'prop-types';
import { ifIphoneX } from 'react-native-iphone-x-helper';

import headerStyles from '../styles/headerStyles';
import BackButton from './BackButton';

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'transparent',
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    height: 42,
    zIndex: 1000
  },
  backButton: {
    position: 'absolute',
    top: ifIphoneX(60, 36),
    left: 10
  },
  title: {
    position: 'absolute',
    top: ifIphoneX(60, 36),
    left: 40,
    right: 40,
    textAlign: 'center'
  }
});

export default class Header extends Component {
  render() {
    const { showBackButton } = this.props;

    return (
      <View style={styles.header}>
        { showBackButton ? <BackButton onPress={this.props.onBackPress} style={styles.backButton} iconStyle={this.props.backButtonIconStyle} /> : null }

        <Text style={[headerStyles.title, styles.title]}>
          {this.props.title}
        </Text>
      </View>
    );
  }
}

Header.propTypes = {
  showBackButton: PropTypes.bool,
  onBackPress: PropTypes.func,
  backButtonIconStyle: PropTypes.any,
  title: PropTypes.string
};
