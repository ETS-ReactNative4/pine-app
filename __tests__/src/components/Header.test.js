import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../../src/components/Header';

describe('Header', () => {
  it('renders correctly with showBackButton set to false', () => {
    const tree = renderer.create(
      <Header showBackButton={false} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with showBackButton set to true', () => {
    const tree = renderer.create(
      <Header showBackButton={true} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
