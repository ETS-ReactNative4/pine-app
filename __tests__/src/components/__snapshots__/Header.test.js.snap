// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with showBackButton set to false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "600",
          "letterSpacing": -0.41,
        },
        Object {
          "left": 40,
          "position": "absolute",
          "right": 40,
          "textAlign": "center",
          "top": 36,
        },
      ]
    }
  />
</View>
`;

exports[`Header renders correctly with showBackButton set to true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 10,
        "opacity": 1,
        "position": "absolute",
        "top": 36,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontSize": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï…‰
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "600",
          "letterSpacing": -0.41,
        },
        Object {
          "left": 40,
          "position": "absolute",
          "right": 40,
          "textAlign": "center",
          "top": 36,
        },
      ]
    }
  />
</View>
`;
